docs/src/main/sphinx/optimizer/pushdown.rst

```diff
@@ -151,6 +151,65 @@ to perform the remaining query processing on a smaller amount of data.
 The specifics for the supported pushdown of table joins varies for each data
 source, and therefore for each connector.
 
+However, there are some generic conditions that must be met in order for a join
+to be pushed down:
+
+* join pushdown must be enabled
+* all predicates that are part of the join must be possible to be pushed down
```

Comment:
nit: although this point implicitly covers it this is such a common question that maybe it deserves a sub-point

```suggestion
* all predicates that are part of the join must be possible to be pushed down
  * predicates that cannot be represented as a series of AND or IN operations cannot be pushed down (e.g. LIKE or OR)
  * predicates that involve function calls
  * predicates that involve complex expressions like type coercion or addition/subtraction etc.
```

Or maybe we don't want to go so deep into implementation details. (I believe nobody should care if pushdown happened or not - if it can the engine will do it. What people actually care about is performance and they substitute pushdown as a proxy for performance).

Reply:
I would prefer to link to a specific section that has more details on predicate pushdown rather than cover it here is part of join pushdown. What do you think?

Reply:
Yeah ... we definitely need a separate section for predicate pushdown.. should we addressed in a separate PR though and then we can add a link to it here

Reply:
There is a PR for predicate pushdown stuff in the works with @Jessie212